blocks/initialization.py

```diff
@@ -147,21 +148,39 @@ def __init__(self, mult=1):
         self.mult = mult
 
     def generate(self, rng, shape):
-        if len(shape) != 2:
-            raise ValueError
-        rows, cols = shape
-        return self.mult * numpy.eye(rows, cols, dtype=theano.config.floatX)
+        if len(shape) == 1:
+            eye_matrix = numpy.eye(1, shape[0], dtype=theano.config.floatX)
+            return self.mult * eye_matrix[0]
+        elif len(shape) == 2:
+            rows, cols = shape
+            eye_matrix = numpy.eye(rows, cols, dtype=theano.config.floatX)
+            return self.mult * eye_matrix
+        else:
+            raise ValueError("Identity initialization only works for 1D and"
+                             " 2D variables")
 
 
 class Orthogonal(NdarrayInitialization):
-    """Initialize a random orthogonal matrix.
+    r"""Initialize a random, approximately orthogonal matrix.
+
+    Only works for 1D and 2D variables.
 
-    Only works for 2D arrays.
+    When used to initialize a 1D parameter vector the elements will
+    be drawn i.i.d. from a normal distribution.
+
+    For 2D matrices we initialalize such that
+    :math:`M M^T \approx I` and :math:`M^T M \approx I` where
+    :math:`I` is an identity matrix of appropriate size.
 
     """
     def generate(self, rng, shape):
-        if len(shape) != 2:
-            raise ValueError
+        if len(shape) not in (1, 2):
+            raise ValueError("Orthogonal initialization only works for 1D and"
+                             " 2D variables")
+
+        if len(shape) == 1:
+            # Special case for 1D variables
+            return rng.randn(*shape).astype(theano.config.floatX)
```

Comment:
Ok, just to recap: 

The way we initialize 1D vectors clearly depends on their usage. If we initialize a vector that is effectively used as a row or column of a matrix, the Orthogonal initialization should be normalized to one. 
If we use it as the diagonal of a connectivity matrix, the (most strict) generalization would  be numpy.ones(shape[0]). I tested this on some small toy tests -- and it gives neither good results, nor is it very intuitive. From that perspective (1. + randn(shape[0])) also looks like a reasonable candidate. 

So when initializing 1D 'weight-vectors' it's neither obvious if they are used coumn/row-wise or as diagonal, Nor is it obvious that the same initialization scheme is well defined and meaningfull even when we know how it is used.

Given all this ambiguity I would propose to keep Identity and Orthogonal as they are: only working for 2D vars, and give an additional argument to LSTM constructor

```
lstm = LSTM(dim=100, weights_init=Orthogonal(), 
diagonal_weights_init=IsotropicGaussian(), biases_init=Constant(0))
```

The default for diagonal_weights_init could be to use the same as weights_init.